cmd/admin-top-locks.go

```diff
@@ -0,0 +1,152 @@
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+  1. Get oldest locks on a Minio server/cluster.
+     $ {{.HelpName}} play/
+
+`,
+}
+
+// lockMessage struct to list lock information.
+type lockMessage struct {
+	Status string           `json:"status"`
+	Lock   madmin.LockEntry `json:"locks"`
+}
+
+func getTimeDiff(timeStamp time.Time) (string, string) {
+	loc, _ := time.LoadLocation("UTC")
+	now := time.Now().In(loc)
+	diff := now.Sub(timeStamp)
+	hours := int(diff.Hours())
+	minutes := int(diff.Minutes()) % 60
+	seconds := int(diff.Seconds()) % 60
+	if hours == 0 {
+		if minutes == 0 {
+			return "Lock", fmt.Sprint(seconds, " seconds")
+		}
+		return "Lock", fmt.Sprint(minutes, " minutes")
+	}
+	return "StaleLock", fmt.Sprint(hours, " hours")
+}
+
+// String colorized oldest locks message.
+func (u lockMessage) String() string {
+	timeFieldMaxLen := 20
+	resourceFieldMaxLen := -1
+	typeFieldMaxLen := 6
+	ownerFieldMaxLen := 20
+
+	lockState, timeDiff := getTimeDiff(u.Lock.Timestamp)
+	return console.Colorize(lockState, newPrettyTable("  ",
+		Field{"Time", timeFieldMaxLen},
+		Field{"Type", typeFieldMaxLen},
+		Field{"Owner", ownerFieldMaxLen},
+		Field{"Resource", resourceFieldMaxLen},
+	).buildRow(timeDiff, u.Lock.Type, u.Lock.Owner, u.Lock.Resource))
+}
+
+// JSON jsonified top oldest locks message.
+func (u lockMessage) JSON() string {
+	u.Status = "success"
+	statusJSONBytes, e := json.Marshal(u)
+	fatalIf(probe.NewError(e), "Unable to marshal into JSON.")
+	return string(statusJSONBytes)
+}
+
+// checkAdminTopLocksSyntax - validate all the passed arguments
+func checkAdminTopLocksSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) == 0 || len(ctx.Args()) > 2 {
+		cli.ShowCommandHelpAndExit(ctx, "top", 1) // last argument is exit code
+	}
+}
+
+func mainAdminTopLocks(ctx *cli.Context) error {
+
+	checkAdminTopLocksSyntax(ctx)
+
+	// Get the alias parameter from cli
+	args := ctx.Args()
+	aliasedURL := args.Get(0)
+
+	// Create a new Minio Admin Client
+	client, err := newAdminClient(aliasedURL)
+	fatalIf(err, "Cannot get a configured admin connection.")
+
+	// Call top locks API
+	entries, e := client.TopLocks()
+	fatalIf(probe.NewError(e), "Cannot get server locks list.")
+
+	console.SetColor("StaleLock", color.New(color.FgRed, color.Bold))
+	console.SetColor("Lock", color.New(color.FgBlue, color.Bold))
+	console.SetColor("Headers", color.New(color.FgGreen, color.Bold))
+
+	// Print
+	printLocks(entries)
+	return nil
+}
+
+func printHeaders() {
+	if !globalJSON {
+		timeFieldMaxLen := 20
+		resourceFieldMaxLen := -1
+		typeFieldMaxLen := 6
+		ownerFieldMaxLen := 20
+		console.Println(console.Colorize("Headers", newPrettyTable("  ",
+			Field{"Time", timeFieldMaxLen},
+			Field{"Type", typeFieldMaxLen},
+			Field{"Owner", ownerFieldMaxLen},
+			Field{"Resource", resourceFieldMaxLen},
+		).buildRow("Time", "Type", "Owner", "Resource")))
+	}
+}
+
+// Prints oldest locks.
+func printLocks(locks madmin.LockEntries) {
+	printHeaders()
```

Comment:
Little nit here, it is just more clear if you move `if !globalJSON` (function names should reflect as much possible actions to avoid bugs in the future)

Reply:
Is this how you want 
```
if !globalJSON {
		printHeaders()
	}
```